<?xml version ="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="RRP">
    <link name="base_link"></link>
    <link name="link0">
        <visual>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link0.dae"/>
            </geometry>
        </visual>
        <collision>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link0.dae"/>
            </geometry>
        </collision>
    </link>
    <joint name="base_joint" type = "fixed">
        <parent link = "base_link"/>
        <child link = "link0"/>
        <origin xyz = "0 0 0"/>
        <dynamics damping="10.0" friction="10.0"/>
    </joint>

    <link name="link1">
        <visual>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link1.dae"/>
            </geometry>
        </visual>
        <collision>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link1.dae"/>
            </geometry>
        </collision>
    </link>

    <joint name="joint_1" type = "revolute">
        <parent link = "link0"/>
        <child link = "link1"/>
        <origin xyz = "0 0 0.3"/>
        <axis xyz="0 0 1"/>
        <limit upper="3" lower="-3" effort="0.1" velocity="0.1"/>
        <dynamics damping="10.0" friction="10.0"/>
    </joint>

    <link name="link2">
        <visual>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link2.dae"/>
            </geometry>
        </visual>
        <collision>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link2.dae"/>
            </geometry>
        </collision>
    </link>

    <joint name="joint_2" type = "revolute">
        <parent link = "link1"/>
        <child link = "link2"/>
        <origin xyz = "0.3 0 0.05"/>
        <axis xyz="0 0 1"/>
        <limit upper="2" lower="1" effort="0.1" velocity="0.1"/>
        <dynamics damping="10.0" friction="10.0"/>
    </joint>

    <link name="link3">
        <visual>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link3.dae"/>
            </geometry>
            <origin xyz = "0 0 0"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_link3.dae"/>
            </geometry>
            <origin xyz = "0 0 0"/>
        </collision>
    </link>

    <joint name="joint_end" type = "prismatic"> 
        <parent link = "link2"/>
        <child link = "link3"/>
        <origin xyz = "0.3 0 0"/>
        <axis xyz="0 0 1"/>
        <limit upper="0.05" lower="-0.1" effort="0.1" velocity="0.1"/>
        <dynamics damping="10.0" friction="10.0"/>
    </joint>

    <link name="end_effector">
        <visual>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_end.dae"/>
            </geometry>
            <origin xyz = "0 0 0"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="file://$(find rl_robot)/meshes/RRP_robot_Extrude_end.dae"/>
            </geometry>
            <origin xyz = "0 0 0"/>
        </collision> 
    </link>

    <joint name="link length" type = "fixed">
        <parent link = "link3"/>
        <child link = "end_effector"/>
        <origin xyz = "0 0 -0.11"/>
        <dynamics damping="10.0" friction="10.0"/>
    </joint>

</robot>